lib/at_host/at_host.c

```diff
@@ -72,28 +80,29 @@ static void cmd_send(struct k_work *work)
 {
 	size_t            chars;
 	char              str[15];
-	static char       buf[AT_MAX_CMD_LEN];
 	enum at_cmd_state state;
 	int               err;
 
 	ARG_UNUSED(work);
 
 	/* Make sure the string is 0-terminated */
-	at_buf[MIN(at_buf_len, AT_MAX_CMD_LEN - 1)] = 0;
+	at_buf[AT_BUF_SIZE - 1] = 0;
 
-	err = at_cmd_write(at_buf, buf, AT_MAX_CMD_LEN, &state);
+	err = at_cmd_write(at_buf, at_buf,
+			   CONFIG_AT_CMD_RESPONSE_MAX_LEN, &state);
```

Comment:
So, here you aren't terminating the string but the entire buffer; and that's also what you are sending to the AT interface. Are you sure that's what we want to do?

Reply:
The termination at line 89 (as of 9de9d79362aa801542ba9d3a8f4ac7504ddc3474) is actually redundant. The proper termination takes place at line 187 (as of 9de9d79362aa801542ba9d3a8f4ac7504ddc3474).

Line 89 only serves as a cheap way to make sure the buffer terminates and the maximum length is upheld, if `cmd_send` ever were to be called while the buffer is not terminated. I can remove it, if desired.

Reply:
You should use `AT_BUF_SIZE` instead of `CONFIG_AT_CMD_RESPONSE_MAX_LEN` here, as they might be different, looking at the logic above.

Or simply use `sizeof(at_buf)` to avoid such mistakes.

Reply:
> I can remove it, if desired.

If it's not needed, we should remove it.

Reply:
> You should use `AT_BUF_SIZE` instead of `CONFIG_AT_CMD_RESPONSE_MAX_LEN` here, as they might be different, looking at the logic above.

The configured limits will always be <= the actual buffer size. The decision to use the configured limits rather than the actual buffer size was conscious on my part. The rationale is: If the actual buffer size is used everywhere, a scenario is possible where e.g. the AT command length limit is decreased/increased in a later commit, and this snowballs into changes in AT response handling behavior, or vice versa. So using the configured limits restricts the scope of errors introduced by a configuration change as much as possible.

That being said, this approach does "waste" the non-configured space for either commands or responses, and `sizeof` is of course more resistant to code changes. Let me know if you still want me to change it, and I will replace with `sizeof` :)

Reply:
I still think `sizeof` would be a neater option in this case.

As you pointed out the buffer size, hence `sizeof(at_buf)`, will always be larger or equal to `CONFIG_AT_CMD_RESPONSE_MAX_LEN`, so I don't see how could we introduce any unexpected failures in case this config is changed in the future. And the API is clear for `at_cmd_write` - we specify buffer capabilities with `buf_len` parameter.

It's good to have a precautions to changes in `CONFIG_AT_CMD_RESPONSE_MAX_LEN `, my point is that this already covered by `AT_BUF_SIZE` definition and buffer declaration, I don't think there's a need to duplicate them.

Reply:
Changed to `sizeof` and removed the unneeded termination